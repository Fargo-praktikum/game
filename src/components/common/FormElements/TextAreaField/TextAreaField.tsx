// import React from "react";
//
// export const TextAreaInput: React.FC<Props> = ({ input, meta, ...restProps }: Props) => {
//     const hasError = meta.touched && meta.error;
//
//     return (
//         <div className={`${styles.formControl} ${hasError ? styles.error : ''}`}>
//             <textarea {...input} {...restProps} />
//             <div>
//                 {hasError && <span>{meta.error}</span> }
//             </div>
//         </div>
//     );
// };

