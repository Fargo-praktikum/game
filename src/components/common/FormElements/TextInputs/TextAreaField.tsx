import React from "react";

import "./TextInputs.scss";
import { FieldProps } from "formik";

type OwnProps = FieldProps<string, any>;
// FIXME: Так линтер не пропускает, наверное стоит отплючить это правило
// export const TextAreaField: React.FC<OwnProps> = ({ field, form, meta, ...props }) => {
export const TextAreaField = ({ field, form, meta, ...props }: Partial<OwnProps>): JSX.Element => {
    const name  = field?.name || "_test";
    const hasError = form?.touched[name] && form.errors[name];

    return (
        <div className={`formControl ${hasError ? "errorInput" : ""}`}>
            <textarea className="textField"
                {...field} {...props} />
            <div className="errorContainer">
                {hasError && (
                    <span className="error">{form?.errors[name]}</span>
                )}
            </div>

        </div>
    );
};


