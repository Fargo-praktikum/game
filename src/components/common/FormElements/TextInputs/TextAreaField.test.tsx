import React from "react";

import renderer from "react-test-renderer";
import { TextAreaField } from "./TextAreaField";

describe("Проверка TextAreaField", () => {
    it("корректная отрисовка", () => {
        const tree = renderer
            .create(<TextAreaField/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("наличие всех свойств", () => {
        const field = { value: "testValue", name: "test", onChange: () => {return;}, onBlur: () => {return;} };
        const tree = renderer.create(<TextAreaField
            field={field}
        />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});



