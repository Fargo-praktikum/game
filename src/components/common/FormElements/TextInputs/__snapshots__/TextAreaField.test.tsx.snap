// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка TextAreaField корректная отрисовка 1`] = `
<div
  className="formControl "
>
  <textarea
    className="textField"
  />
  <div
    className="errorContainer"
  />
</div>
`;

exports[`Проверка TextAreaField наличие всех свойств 1`] = `
<div
  className="formControl "
>
  <textarea
    className="textField"
    name="test"
    onBlur={[Function]}
    onChange={[Function]}
    value="testValue"
  />
  <div
    className="errorContainer"
  />
</div>
`;
